v2/yarpcgrpc/outbound.go

```diff
@@ -46,83 +42,55 @@ import (
 // http://www.grpc.io/docs/guides/wire.html#user-agents
 const UserAgent = "yarpc-go/" + yarpc.Version
 
-var _ transport.UnaryOutbound = (*Outbound)(nil)
+var _ yarpc.UnaryOutbound = (*Outbound)(nil)
 
-// Outbound is a transport.UnaryOutbound.
+// Outbound sends YARPC requests over gRPC. It is recommended that services use
+// a single HTTP dialer to construct all HTTP outbounds, ensuring efficient
+// sharing of resources across the different outbounds.
 type Outbound struct {
-	once        *lifecycle.Once
-	lock        sync.Mutex
-	t           *Transport
-	peerChooser peer.Chooser
-	options     *outboundOptions
-}
+	// Chooser is a peer chooser for outbound requests.
+	Chooser yarpc.Chooser
 
-func newSingleOutbound(t *Transport, address string, options ...OutboundOption) *Outbound {
-	return newOutbound(t, peerchooser.NewSingle(hostport.PeerIdentifier(address), t), options...)
-}
+	// Dialer is an alternative to specifying a Chooser.
```

Comment:
Might be worth adding a note that the `Chooser` will be prioritized over the `Dialer`, if both are set.